### ISSUE TITLE
Default to Forward KL?

### ISSUE THREAD

**Issue by hank0316** [idanshen/Self-Distillation#5]
Hi @idanshen,

Thanks for open-sourcing the training code.

I noticed that the default value of `alpha` in `DistilConfig` is set to `0`:

https://github.com/idanshen/Self-Distillation/blob/1825ef3074b6289fffe285145ce84522c1f059d3/distil_config.py#L486-L491

And is not overridden in `main.py`:
https://github.com/idanshen/Self-Distillation/blob/1825ef3074b6289fffe285145ce84522c1f059d3/main.py#L62-L90

In the trainer implementation, it seems that when `alpha = 0`, the trainer will use **forward** KL, instead of reverse KL (`eq 1` in your paper):

https://github.com/idanshen/Self-Distillation/blob/1825ef3074b6289fffe285145ce84522c1f059d3/distil_trainer.py#L1659-L1662

Is this the intended default behavior? Not sure if I misunderstood the loss calculation logics.

Looking forward to your clarification!

**Comment by idanshen** [comment]
After some investigation, we've found that all the results in our paper were produced using on-policy sampling, but per-token forward KL loss (similar to the [GKD paper](https://arxiv.org/abs/2306.13649)). Therefore, this is the default argument in this repo, and we will update the arXiv version soon with clarification.
